src/network/netlink.rs

```diff
@@ -0,0 +1,407 @@
+use std::{
+    net::{Ipv4Addr, Ipv6Addr},
+    os::unix::prelude::RawFd,
+};
+
+use crate::{
+    error::{ErrorWrap, NetavarkError, NetavarkResult},
+    wrap,
+};
+use log::trace;
+use netlink_packet_route::{
+    nlas::link::{Info, InfoData, InfoKind, Nla},
+    AddressMessage, LinkMessage, NetlinkHeader, NetlinkMessage, NetlinkPayload, RouteMessage,
+    RtnlMessage, AF_INET, AF_INET6, IFF_UP, NLM_F_ACK, NLM_F_CREATE, NLM_F_DUMP, NLM_F_EXCL,
+    NLM_F_REQUEST, RTN_UNICAST, RTPROT_STATIC, RT_SCOPE_UNIVERSE, RT_TABLE_MAIN,
+};
+use netlink_sys::{protocols::NETLINK_ROUTE, SocketAddr};
+
+pub struct Socket {
```

Comment:
add  #[derive(Clone )] please

Reply:
Why do you need to clone it? The problem is that you end up having the objects that use the same underlying socket and it will result in problems when you start using it in parallel.

I can definitely add it but it must be carefully used so it would not cause bugs.

Reply:
AFAIK copy is just clone without having to explicitly call .clone() which means that you start cloning without realizing it which is IMO a bad thing

Reply:
yeah, lets skip this.